datashader/transfer_functions/__init__.py

```diff
@@ -375,8 +346,10 @@ def _colorize(agg, color_key, how, span, min_alpha, name):
         # i.e. a 0 will be fully transparent, but any non-zero number will
         # be clipped to the span range and have min-alpha applied
         offset = np.array(span, dtype=data.dtype)[0]
-        if offset <= 0:
+        if np.nanmin(total) == 0 and total.dtype.kind == 'u':
             mask = mask | (total <= 0)
+            total = np.where(~mask, total, np.nan)
+        masked_clip_2d(total, mask, *span)
```

Comment:
Clip values outside the provided span ensuring their alpha values do not over/underflow during interpolation.